File: examples/go/test_cross_module_calls.go
// Test file for cross-module call tracking in Go
package main

import (
	"fmt"
	"github.com/example/project/config"
	"github.com/example/project/utils/helper"
)

// InitConfigFile initializes the configuration file
func InitConfigFile() {
	// This should be tracked as calling config.InitGlobalDirs
	config.InitGlobalDirs()

	// This should be tracked as calling utils/helper.ProcessData
	helper.ProcessData()
}

// InitGlobalDirs initializes global directories (local function)
func InitGlobalDirs() {
	fmt.Println("Initializing global directories locally")
}

// ProcessLocalData processes data locally
func ProcessLocalData() {
	// This calls the local InitGlobalDirs
	InitGlobalDirs()

	// This calls the imported config.InitGlobalDirs
	config.InitGlobalDirs()
}

File: examples/php/test_cross_module_calls.php
<?php
/**
 * Test file for cross-module call tracking in PHP
 */

namespace App\Config;

use App\Utils\Helper;
use App\Services\Auth\UserManager;

class ConfigManager {
    /**
     * Initialize configuration file
     */
    public function initConfigFile() {
        // This should be tracked as calling App\Init\GlobalDirs::initGlobalDirs
        \App\Init\GlobalDirs::initGlobalDirs();

        // This should be tracked as calling Helper::processData
        Helper::processData();

        // This should be tracked as calling UserManager::createUser
        UserManager::createUser("test@example.com");
    }

    /**
     * Initialize global directories (local method)
     */
    public function initGlobalDirs() {
        echo "Initializing global directories locally\n";
    }

    /**
     * Process local data
     */
    public function processLocalData() {
        // This calls the local method
        $this->initGlobalDirs();

        // This calls the static method from another namespace
        \App\Init\GlobalDirs::initGlobalDirs();
    }
}